src/main/java/com/rcv/TabulatorSession.java

```diff
@@ -188,8 +188,14 @@ private boolean setUpLogging(ContestConfig config) {
       Logger.log(Level.INFO, "Reading cast vote record file: %s...", cvrPath);
       // the CVRs parsed from this source
       try {
-        List<CastVoteRecord> cvrs =
-            new StreamingCVRReader(config, source).parseCVRFile(castVoteRecords, precinctIDs);
+        List<CastVoteRecord> cvrs;
+        if(source.getProvider().equals("CDF")) {
+          CommonDataFormatReader reader = new CommonDataFormatReader(cvrPath);
+          cvrs = reader.parseCVRFile(castVoteRecords);
```

Comment:
I just noticed that this API is weird. The caller passes in a collection, and then for every CVR that the method parses, it adds it both to the passed-in collection and to a new collection that it returns.

It would be more intuitive if the caller didn't pass in the collection. When the method returns `cvrs`, we can just call `castVoteRecords.addAll(cvrs)` here.

Reply:
Yeah, totally agree.  I think it got to the current state incrementally with no real problem, but it should be easy enough to fix.